### PR TITLE
Add ipadm_addr module

### PR DESCRIPTION
**Pull request by xen0l**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
ipadm_addr

##### ANSIBLE VERSION
```
2.3
```

##### SUMMARY
Module for managing IP address configuration on illumos/Solaris systems.

### CODE REVIEW

File: lib/ansible/modules/network/illumos/ipadm_addr.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+    if out:
+        result['stdout'] = out
+    if err:
+        result['stderr'] = err
+
+    module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addr.py]: please use explicit imports

File: lib/ansible/modules/network/illumos/ipadm_addr.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+main()
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addr.py]: please wrap in `if` block

**Review by privateip — CHANGES_REQUESTED** [review]
* missing RETURN doc string
* missing METADATA
* please consider adding unit test cases

**Comment by xen0l** [comment]
ready_for_review

**Comment by xen0l** [comment]
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

File: lib/ansible/modules/network/illumos/ipadm_addr.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+DOCUMENTATION = '''
+---
+module: ipadm_addr
+short_description: Manage IP addresses on an interface.
+description:
+    - Create/delete static/dynamic IP addresses on network interfaces on Solaris/illumos systems.
+    - Up/down static/dynamic IP addresses on network interfaces on Solaris/illumos systems.
+    - Manage IPv6 link-local addresses on network interfaces on Solaris/illumos systems.
+version_added: "2.1"
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/ipadm_addr.py]: VERSION_ADDED should be "2.3"

**Review by privateip — CHANGES_REQUESTED** [review]

**Comment by xen0l** [comment]
@privateip I bumped version to 2.3

**Pull request merged**